src/naming/main.cairo

```diff
@@ -316,6 +377,137 @@ mod Naming {
+                contract_address: self._pricing_contract.read()
+            }
+                .compute_renew_price(domain_len, days);
+            // compute domain cost in altcoin
+            let price_in_altcoin = self
+                .get_altcoin_price(quote.into(), price_in_eth.try_into().unwrap());
+            self
+                .pay_domain(
+                    domain_len,
+                    altcoin_addr,
+                    price_in_altcoin,
+                    now,
+                    days,
+                    domain,
+                    sponsor,
+                    discount_id
+                );
+            self.emit(Event::SaleMetadata(SaleMetadata { domain, metadata }));
+            // find new domain expiry
+            let new_expiry = if domain_data.expiry <= now {
+                now + 86400 * days.into()
+            } else {
+                domain_data.expiry + 86400 * days.into()
+            };
+            // 25*365 = 9125
+            assert(new_expiry <= now + 86400 * 9125, 'purchase too long');
+            assert(days >= 6 * 30, 'purchase too short');
+
+            let data = DomainData {
+                owner: domain_data.owner,
+                resolver: domain_data.resolver,
+                address: domain_data.address,
+                expiry: new_expiry,
+                key: domain_data.key,
+                parent_key: 0,
+            };
+            self._domain_data.write(hashed_domain, data);
+            self.emit(Event::DomainRenewal(DomainRenewal { domain, new_expiry }));
+        }
+
+        fn auto_renew_altcoin(
+            ref self: ContractState,
+            domain: felt252,
+            days: u16,
+            sponsor: ContractAddress,
+            discount_id: felt252,
+            metadata: felt252,
+            altcoin_addr: ContractAddress,
+            price_in_altcoin: u256,
+        ) {
+            let now = get_block_timestamp();
+            let hashed_domain = self.hash_domain(array![domain].span());
+            let domain_data = self._domain_data.read(hashed_domain);
+
+            // check caller is a whitelisted altcoin auto renewal contract
+            assert(
+                self._whitelisted_renewal_contracts.read(get_caller_address()),
+                'Caller not whitelisted'
+            );
+
+            // we need a u256 to be able to perform safe divisions
+            let domain_len = self.get_chars_len(domain.into());
+            self
+                .pay_domain(
+                    domain_len,
+                    altcoin_addr,
+                    price_in_altcoin,
+                    now,
+                    days,
+                    domain,
+                    sponsor,
+                    discount_id
+                );
+            self.emit(Event::SaleMetadata(SaleMetadata { domain, metadata }));
+            // find new domain expiry
+            let new_expiry = if domain_data.expiry <= now {
```

Comment:
I think we could drop these two expiry check because the auto renew contract already ensures you can't do something abusive (your domain should expire soon and we can renew only for one year)